tools/importPlanGoals.js

```diff
@@ -0,0 +1,185 @@
+              allTopics.forEach((topicPipeRoles) => {
+                const topic = topicPipeRoles.split('|')[0];
+                const roles = topicPipeRoles.split('|')[1];
+                const trimmedTopic = topic.trim();
+                if (trimmedTopic !== '') {
+                  const topicId = getItemId(cleanTopics, trimmedTopic);
+                  if (roles) {
+                    const rolesArr = roles.split(',');
+                    rolesArr.forEach((role) => {
+                      const trimmedRole = role.trim();
+                      const roleId = getItemId(cleanRoles, trimmedRole);
+                      // associate topic with roles
+                      if (!cleanRoleTopics.some((e) => e.roleId === roleId
+                                                      && e.topicId === topicId)) {
+                        cleanRoleTopics.push({ roleId, topicId });
+                      }
+                    });
+                  }
+                  // Add topic to junction with goal
+                  currentGoalId = cleanGoals.length + currentGoals.length - 1; // -1 because there is a dummy element at index 0
+                  cleanTopicGoals.push({ topicId, goalId: currentGoalId });
+                }
+              });
+            } else // it's either "timeframe" or "status"
+            // both "timeframe" and "status" column names will be reused as goal's object keys
+            if (currentGoalName !== '') {
+              currentGoal[column] = el[key].trim();
+              const goalNum = goalColumn[1].slice(0, 1); // represents a goal number from 1 to 5
+              currentGoals[goalNum] = { ...currentGoals[goalNum], ...currentGoal };
+            }
+          }
+        }
+      });
+
+      // after each row
+      let goalId;
+      let grantId;
+
+      currentGoals.forEach((goal) => {
+        cleanGoals.push(goal);
+        goalId = cleanGoals.length;
+        currentGranteeId = getItemId(cleanGrantees, currentGrantee.trim());
+        currentGrants.forEach((grant) => {
+          const fullGrant = { number: grant.trim(), granteeId: currentGranteeId };
+          if (!cleanGrants.some((e) => e.granteeId === fullGrant.granteeId
+                            && e.number === fullGrant.number)) {
+            cleanGrants.push(fullGrant);
+          }
+          grantId = cleanGrants.findIndex((g) => g.number === fullGrant.number) + 1;
+          const plan = { granteeId: currentGranteeId, grantId, goalId };
+          if (!cleanGrantGoals.some((e) => e.granteeId === currentGranteeId
+                            && e.grantId === grantId
+                            && e.goalId === goalId)) {
+            cleanGrantGoals.push(plan);
+          }
+        });
+      });
+    });
+
+    // The data has been prepared. Insert it into the database
+    await Role.bulkCreate(cleanRoles.map((role, index) => (
+      { id: index + 1, name: role })),
+    {
+      updateOnDuplicate: ['name', 'updatedAt'],
```

Comment:
This makes it seem like running the script multiple times won't create multiple roles with the same name. Is that correct?